File: LeetCode/Python/1071_Greatest_Common_Divisor_of_Strings.py
# 1071. Greatest Common Divisor of Strings
# https://leetcode.com/problems/greatest-common-divisor-of-strings/


from math import gcd


class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if str1 + str2 != str2 + str1:
            return ""
        return str1[: gcd(len(str1), len(str2))]

File: LeetCode/JavaScript/1071-Greatest-Common-Divisor-of-Strings.js
// 1071. Greatest Common Divisor of Strings
// https://leetcode.com/problems/greatest-common-divisor-of-strings/description/

/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */

var gcdOfStrings = function (str1, str2) {
  if (str1 + str2 !== str2 + str1) return '';
  let a = str1.length,
    b = str2.length;
  while (b) {
    const temp = b;
    b = a % b;
    a = temp;
  }
  return str1.substring(0, a);
};
